### PR TITLE
fix: DH-17242: Partition Table Pickers not partitioning table properly

### PR DESCRIPTION
**Pull request by AkshatJawne**
Resolves https://github.com/deephaven/web-client-ui/issues/2066

This PR is solely to solve the bugs of the partition table; first that the widget was not appearing, and once appearing, the partition selectors not working properly. The `PartitionedTable` design changes and additional functionality specified in https://github.com/deephaven/web-client-ui/issues/2079 to be brought up here https://github.com/deephaven/web-client-ui/pull/2110

### CODE REVIEW

File: packages/iris-grid/src/IrisGridPartitionSelector.tsx [hunk 1]
```diff
@@ -327,7 +321,9 @@ class IrisGridPartitionSelector extends Component<
     ));
     return (
       <div className="iris-grid-partition-selector">
-        <div className="table-name">Partitioned Table</div>
+        <div className="table-name">
+          {model.isPartitionRequired ? 'Partitioned Table' : ''}
+        </div>
```
Comment (mattrunyon) [inline packages/iris-grid/src/IrisGridPartitionSelector.tsx:326]: This leaves a weird divider on the left side when it's not a partitioned table. Why the empty string?

It looks like the epic ticket says to just remove this label

**Review by mattrunyon — CHANGES_REQUESTED** [review]

**Comment by mattrunyon** [comment]
I think your e2e tests might be failing due to a core change. Looks like it's failing to apply date filters

**Comment by AkshatJawne** [comment]
Hmm interesting, I guess I will try and pull the newest version of main?

**Comment by AkshatJawne** [comment]
I do see that is due to a screenshot, but I thought that it may be due to the change in the Partition Table label that I made here

**Review by mattrunyon — CHANGES_REQUESTED** [review]
Ok so looks like `IrisGridPartitionSelector.scss` needs some modification. There's still the divider line to the left of the buttons on the left side of this image
image

Looks like you need to change `.partition-button-group` for `border-style` to `border-right-style: solid` . And then `padding` to `padding-right: $spacer-2`

**Comment by mattrunyon** [comment]
Also, if this is the PR that fixes the linked ticket (there are 2 and I think this is the real one as the other PR just includes these commits right now) then change the title to `fix: DH-17242: Partition...`. That way when we update the web package versions in enterprise, we see the Jira ticket in the changelog and know to update its status

File: packages/iris-grid/src/IrisGridPartitionSelector.scss [hunk 2]
```diff
@@ -18,8 +18,7 @@
   }
   .partition-button-group {
     display: flex;
-    border: 1px var(--dh-color-hr);
-    border-style: none solid;
+    border-right: 1px solid var(--dh-color-hr);
     padding: 0 $spacer-2;
```
Comment (mattrunyon) [inline packages/iris-grid/src/IrisGridPartitionSelector.scss:22]: Need to change the `padding` too. There's too much gap on the left with this padding. Should just be `padding-right: $spacer-2`

**Review by mattrunyon — CHANGES_REQUESTED** [review]

**Review by mattrunyon — APPROVED** [review]

**Comment by AkshatJawne** [comment]
Will hold off on merging until we can add filter prop to the Picker component, to avoid having to set the viewport on each render to 0,50 -- @bmingles mentioned that will have to wait for #2077 to merge before making these changes.

**Review by mattrunyon — CHANGES_REQUESTED** [review]
Switching to request changes so it doesn't get accidentally merged

**Comment by AkshatJawne** [comment]
Closing this PR given that we are no longer opting to manually set the viewport to 0,50 -- and we ultimately did not add a filter prop to Picker, but rather changed how the options are populated in the pickers.

Refer to https://github.com/deephaven/web-client-ui/pull/2110 for all details and updated changes

**Pull request closed without merging**